### PR TITLE
Bulk request optimalization

### PR DESCRIPTION
**Pull request by bynjamin**
DEPENDS ON yoroi-compatibility BRANCH!!!

### CODE REVIEW

File: app/frontend/helpers/hashCode.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const hashCode = (s) => {
```
Comment (refi93) [inline app/frontend/helpers/hashCode.js]: please cite source if you found it somewhere. Is it some standard hash algorithm? If so, please call the function to reflect that.

**Comment by refi93** [comment]
you probably forgot to delete  `app/tests/src/common/singleAddressesMock.js`

File: app/tests/src/common/bulkSummaryRequests.js [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    'Ae2tdPwUPEZLYjtsKf6xjC5TCmD3RKS5BXMY3bhPv9P8X7QATo3dLzVLU2m',
+    'Ae2tdPwUPEZEwDwRoBzJt3WwhpdEZXzQZTMx7xEjUGRqN2TziD7dj87bTnz',
+    'Ae2tdPwUPEZL7TdN2Ugr3isi3m5UoH6nXnMZA63VuYBBsbTiqALv3UMM2bS',
+    'Ae2tdPwUPEZ1j7ShGjWmWJNmtN9neEYgMcixuguum8cidaRnGERGh8QMDGb',
+    'Ae2tdPwUPEZHszSLNV6L8qAFE1Jh7Q8u5z3h4NU3NRReAn4e47zhDgJTnRC',
+    'Ae2tdPwUPEZN3GEFq8FoMzsSchMRwjcwzUoHsy3ARHUHQQHZxYzeDXtAkir',
+    'Ae2tdPwUPEZD9a9iktQ4YRVU8tSUE5KgUWJAWcVhNrh8G3xEXbd2ZJBH7Jx',
+  ],
+  [
+    'Ae2tdPwUPEZ6RUCnjGHFqi59k5WZLiv3HoCCNGCW8SYc5H9srdTzn1bec4W',
```
Comment (refi93) [inline app/tests/src/common/bulkSummaryRequests.js]: this seems weird - why is there a request for so many addresses? I would expect that each of the summary requests happens in a bulk of `GAP_LIMIT` addresses, but this is far more than 20

File: app/frontend/helpers/hashCode.js (hunk 1 above)
Comment (bynjamin) [reply app/frontend/helpers/hashCode.js]: https://gist.github.com/iperelivskiy/4110988

**Pull request merged**